File: WPF/Controls/NavigationPage.vb
Imports Ks.Common.MVVM

Namespace Common.Controls

    <ComponentModel.DesignTimeVisible(False)>
    Public Class NavigationPage
        Inherits Page
        Implements INavigationView

        Shared Sub New()
            DefaultStyleKeyProperty.OverrideMetadata(GetType(NavigationPage), New FrameworkPropertyMetadata(GetType(NavigationPage)))
        End Sub

        Private Property INavigationView_Content As Page Implements INavigationView.Content
            Get
                Return DirectCast(Me.Content, Page)
            End Get
            Set(ByVal Value As Page)
                Me.Content = Value
            End Set
        End Property

    End Class

End Namespace

File: WPF/Controls/NavigationPage.cs
using System.Windows;
using Ks.Common.MVVM;

namespace Ks.Common.Controls
{
    [System.ComponentModel.DesignTimeVisible(false)]
    public class NavigationPage : Page, INavigationView
    {
        static NavigationPage()
        {
            DefaultStyleKeyProperty.OverrideMetadata(typeof(NavigationPage), new FrameworkPropertyMetadata(typeof(NavigationPage)));
        }

        Page INavigationView.Content
        {
            get
            {
                return (Page) this.Content;
            }
            set
            {
                this.Content = value;
            }
        }
    }
}
